File: iosApp/deadly/Feature/Player/AutoAdvanceTakeover.swift
import SwiftUI

/// ADR-0010 — full-screen "Up Next" end-of-show takeover: the full-player
/// counterpart to the docked `AutoAdvanceOverlay`. While the countdown runs and
/// the player is open, the player is replaced by a preview of the NEXT show —
/// the same large artwork / date / venue layout as the now-playing view, under
/// an "Up Next in Ns" header, with Play now / Cancel. Mirrors web's HeaderPlayer
/// takeover. Fed by `AutoAdvanceCoordinator.countdown`; renders nothing when
/// there's no countdown.
struct AutoAdvanceTakeover: View {
    @Environment(\.appContainer) private var container

    var body: some View {
        if let countdown = container.autoAdvanceCoordinator.countdown {
            let show = countdown.nextShow
            ZStack {
                Color(.systemBackground).ignoresSafeArea()

                VStack(spacing: 0) {
                    // "Up Next in Ns" header — the countdown the user watches tick.
                    Text("UP NEXT IN \(countdown.secondsRemaining)s")
                        .font(.subheadline.weight(.semibold))
                        .tracking(2)
                        .foregroundStyle(DeadlyColors.primary)

                    Spacer().frame(height: 24)

                    // Large cover art of the next show — same prominence (300pt)
                    // and shadow as the now-playing player's artwork.
                    ShowArtwork(
                        recordingId: show.bestRecordingId,
                        imageUrl: show.coverImageUrl,
                        size: 300,
                        cornerRadius: DeadlySize.carouselCornerRadius
                    )
                    .shadow(color: .black.opacity(0.4), radius: 20, y: 10)

                    Spacer().frame(height: 32)

                    // Next show date / venue / location.
                    VStack(spacing: 6) {
                        Text(show.date)
                            .font(.title2.weight(.bold))
                            .foregroundStyle(.primary)
                            .lineLimit(2)
                            .multilineTextAlignment(.center)
                        if !show.venue.name.isEmpty {
                            Text(show.venue.name)
                                .font(.subheadline)
                                .foregroundStyle(.secondary)
                                .lineLimit(1)
                        }
                        if !show.location.displayText.isEmpty {
                            Text(show.location.displayText)
                                .font(.footnote)
                                .foregroundStyle(.secondary.opacity(0.8))
                                .lineLimit(1)
                        }
                    }

                    Spacer().frame(height: 32)

                    // Actions in the transport's place: start now, or stay put.
                    Button {
                        container.autoAdvanceCoordinator.playNow()
                    } label: {
                        Text("Play now")
                            .font(.subheadline.weight(.bold))
                            .padding(.horizontal, 32)
                            .padding(.vertical, 12)
                            .background(DeadlyColors.primary)
                            .foregroundStyle(.black)
                            .clipShape(Capsule())
                    }

                    Spacer().frame(height: 12)

                    Button {
                        container.autoAdvanceCoordinator.cancel()
                    } label: {
                        Text("Cancel")
                            .font(.subheadline)
                            .foregroundStyle(.secondary)
                            .padding(.horizontal, 24)
                            .padding(.vertical, 10)
                            .overlay(Capsule().stroke(Color.secondary.opacity(0.4)))
                    }
                }
                .padding(.horizontal, 24)
            }
            .transition(.opacity)
        }
    }
}

File: androidApp/app/src/main/java/com/grateful/deadly/playback/AutoAdvanceTakeover.kt
package com.grateful.deadly.playback

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.grateful.deadly.core.design.component.ShowArtwork

/**
 * ADR-0010 — end-of-show "Up Next" takeover. The full-player counterpart to the
 * docked [AutoAdvanceOverlay]: while the countdown runs and the player screen is
 * open, the whole player is replaced by a preview of the NEXT show — the same
 * large cover-art / date / venue layout as the now-playing player, under an
 * "Up Next in Ns" header, with Play now / Cancel. Mirrors web's HeaderPlayer
 * takeover ([ui/.../HeaderPlayer.tsx]). Fed by [AutoAdvanceCoordinator.Countdown].
 */
@Composable
fun AutoAdvanceTakeover(
    countdown: AutoAdvanceCoordinator.Countdown,
    onPlayNow: () -> Unit,
    onCancel: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val show = countdown.nextShow
    Surface(
        modifier = modifier.fillMaxSize(),
        color = MaterialTheme.colorScheme.background,
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
        ) {
            // "Up Next in Ns" header — the countdown the user is watching tick.
            Text(
                text = "UP NEXT IN ${countdown.secondsRemaining}s",
                style = MaterialTheme.typography.labelLarge,
                fontWeight = FontWeight.SemiBold,
                letterSpacing = 2.sp,
                color = MaterialTheme.colorScheme.primary,
            )

            Spacer(Modifier.height(24.dp))

            // Large cover art of the next show — same prominence as the player's.
            ShowArtwork(
                recordingId = show.bestRecordingId,
                imageUrl = show.coverImageUrl,
                contentDescription = show.date,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(360.dp),
            )

            Spacer(Modifier.height(24.dp))

            // Next show date / venue / location.
            Text(
                text = show.date,
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            if (show.venue.name.isNotBlank()) {
                Spacer(Modifier.height(4.dp))
                Text(
                    text = show.venue.name,
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    textAlign = TextAlign.Center,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
            if (show.location.displayText.isNotBlank()) {
                Text(
                    text = show.location.displayText,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.7f),
                    textAlign = TextAlign.Center,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }

            Spacer(Modifier.height(32.dp))

            // Actions in the transport's place: start now, or stay on this show.
            Button(
                onClick = onPlayNow,
                contentPadding = PaddingValues(horizontal = 32.dp, vertical = 12.dp),
            ) {
                Text("Play now", style = MaterialTheme.typography.labelLarge)
            }
            Spacer(Modifier.height(12.dp))
            OutlinedButton(
                onClick = onCancel,
                contentPadding = PaddingValues(horizontal = 24.dp, vertical = 10.dp),
            ) {
                Text("Cancel", style = MaterialTheme.typography.labelLarge)
            }
        }
    }
}
